test/Interpreter/bridged_casts_folding.swift

```diff
@@ -302,11 +302,9 @@ Tests.test("NSDictionary -> Swift (Dictionary). Crashing Test Cases") {
     expectNil(dictOpt)
   }
 
-  // NOTE: This is preserving current behavior. We should probably have a better
-  // error here like in the other crashes.
-  //
   // CHECK-LABEL: [ RUN      ] BridgedCastFolding.NSDictionary -> Swift (Dictionary). Crashing Test Cases
-  // CHECK: stderr>>> OK: saw expected "crashed: sigill"
+  // CHECK: stderr>>> Could not cast value of type '{{.*}}' (0x{{[0-9a-f]*}}) to 'NSDictionary' (0x{{[0-9a-f]*}}).
```

Comment:
Question. Can you just eliminate NSDictionary as well. Bonus points if you want to fix the other class names in the file. Please do at least the first.

Funny coincidence!

Reply:
I think hardwiring NSDictionary is fully justified here—it comes from the _ObjectiveCType of `[Int: Int]`’s _ObjectiveCBridgeable conformance, and is fully under our control. There is no way it can/should ever change. (While the dynamic type of the input could change from one platform/OS release to the next.)

Reply:
(Resolved in #23185)